pkg/operator/configobserver/apiserver/observe_tlssecurityprofile.go

```diff
@@ -24,33 +24,46 @@ type APIServerLister interface {
 	PreRunHasSynced() []cache.InformerSynced
 }
 
-// ObserveTLSSecurityProfile observes APIServer.Spec.TLSSecurityProfile field and sets
-// the ServingInfo.MinTLSVersion, ServingInfo.CipherSuites fields of observed config
-func ObserveTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	return innerTLSSecurityProfileObservations(genericListers, recorder, existingConfig, []string{"servingInfo", "minTLSVersion"}, []string{"servingInfo", "cipherSuites"})
+type tlsSecurityProfileObserver struct {
+	minTLSVersionPath []string
+	cipherSuitesPath  []string
 }
 
-// ObserveTLSSecurityProfileToArguments observes APIServer.Spec.TLSSecurityProfile field and sets
-// the tls-min-version and tls-cipher-suites fileds of observedConfig.apiServerArguments
-func ObserveTLSSecurityProfileToArguments(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	return innerTLSSecurityProfileObservations(genericListers, recorder, existingConfig, []string{"apiServerArguments", "tls-min-version"}, []string{"apiServerArguments", "tls-cipher-suites"})
+// NewTLSSecurityProfileObserver creates a new observer for the tls tlsSecurityProfile field,
+// it observes to the `servingInfo.minTLSVersion` and `servingInfo.cipherSuites`
```

Comment:
it observers `APIServer.Spec.TLSSecurityProfile` and sets `servingInfo.minTLSVersion/servingInfo.cipherSuites` or `apiServerArguments.tls-min-version/apiServerArguments.tls-cipher-suites` based on the `useArguments` param